### ISSUE TITLE
plugin uses toJSON to set state, but this causes issues when overriding

### ISSUE THREAD

**Issue by McWatt** [magalhas/backbone-react-component#83]
This code uses the `toJSON` method internally for setting state, but that method is for JSON stringification. Look on line 278 of backbone-react-component.js. 

```javascript
      var newState = modelOrCollection.toJSON ? modelOrCollection.toJSON() : modelOrCollection;
```
This causes issues when overriding the toJSON method on models/collections. For example, I had to do this on a collection to save to a legacy api:

```javascript
		toJSON: function(options) {
			var col = this.map(function(model){ return model.toJSON(options); });
			return {
				'id' : 100,
				'collection' : col
			}
		},

```

**Comment by magalhas** [comment]
Hi,

Do you have any suggestion regarding an alternative to `toJSON`?! Maybe directly accessing the models/attributes?

**Comment by McWatt** [comment]
Yep, the attributes would be accessed directly. Here is a conversation in the backbone repo where this topic is discussed: https://github.com/jashkenas/backbone/issues/2134

**Comment by McWatt** [comment]
I hacked out a fix for my local dev. This hasn't really been tested but so far appears to be working.

```javascript
    getAttributes: function (modelOrCollection){
      var attrs = [];

      // if a collection, get the attributes of each, otherwise return modelOrCollection
      if (modelOrCollection instanceof Backbone.Collection) {
        
        for (var i = 0; i < modelOrCollection.models.length; i++) {
          attrs.push(modelOrCollection.models[i].attributes);
        }

        return attrs;
        
      } else {
        return modelOrCollection
      }
      
    },
```
Then call the helper from ```setState```:
```javascript
      var newState = this.getAttributes(modelOrCollection);
```

**Comment by magalhas** [comment]
Glad you were able to find a solution. I'll take a look at your solution and check if I can incorporate it into the lib. Thank you :+1:

**Comment by magalhas** [comment]
Care to craft a PR so we can fit this into a patch release? The solution seems fine, and if anyone is relying on the `toJSON` hook to change what goes into the UI there are many alternatives. Also this will probably improve performance.

**Comment by fifigyuri** [comment]
The solution by @McWatt works for me, it seems to be pretty simple.

**Comment by magalhas** [comment]
Fixed by #86
